spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/config/RabbitConfiguration.java

```diff
@@ -0,0 +1,125 @@
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.retry.support.RetryTemplate;
+
+/**
+ * Configuration for {@link RabbitTemplate} and {@link CachingConnectionFactory}.
+ *
+ * @author Chris Bono
+ * @since 3.2
+ */
+@Configuration(proxyBeanMethods = false)
+@EnableConfigurationProperties(RabbitProperties.class)
+public class RabbitConfiguration {
+
+	@Bean
+	@ConditionalOnMissingBean
+	RabbitConnectionFactoryBeanConfigurer rabbitConnectionFactoryBeanConfigurer(RabbitProperties properties,
+			ResourceLoader resourceLoader, ObjectProvider<CredentialsProvider> credentialsProvider,
+			ObjectProvider<CredentialsRefreshService> credentialsRefreshService) {
+		RabbitConnectionFactoryBeanConfigurer configurer = new RabbitConnectionFactoryBeanConfigurer();
+		configurer.setRabbitProperties(properties);
+		configurer.setResourceLoader(resourceLoader);
+		configurer.setCredentialsProvider(credentialsProvider.getIfUnique());
+		configurer.setCredentialsRefreshService(credentialsRefreshService.getIfUnique());
+		return configurer;
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	CachingConnectionFactoryConfigurer rabbitConnectionFactoryConfigurer(RabbitProperties rabbitProperties,
+			ObjectProvider<ConnectionNameStrategy> connectionNameStrategy) {
+		CachingConnectionFactoryConfigurer configurer = new CachingConnectionFactoryConfigurer();
+		configurer.setRabbitProperties(rabbitProperties);
+		configurer.setConnectionNameStrategy(connectionNameStrategy.getIfUnique());
+		return configurer;
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(ConnectionFactory.class)
+	CachingConnectionFactory rabbitConnectionFactory(
+			RabbitConnectionFactoryBeanConfigurer rabbitConnectionFactoryBeanConfigurer,
+			CachingConnectionFactoryConfigurer rabbitCachingConnectionFactoryConfigurer,
+			ObjectProvider<ConnectionFactoryCustomizer> connectionFactoryCustomizers) throws Exception {
+
+		RabbitConnectionFactoryBean connectionFactoryBean = new RabbitConnectionFactoryBean();
+		rabbitConnectionFactoryBeanConfigurer.configure(connectionFactoryBean);
+		connectionFactoryBean.afterPropertiesSet();
+		com.rabbitmq.client.ConnectionFactory connectionFactory = connectionFactoryBean.getObject();
+		connectionFactoryCustomizers.orderedStream()
+				.forEach((customizer) -> customizer.customize(connectionFactory));
+
+		CachingConnectionFactory cachingConnectionFactory = new CachingConnectionFactory(connectionFactory);
+		rabbitCachingConnectionFactoryConfigurer.configure(cachingConnectionFactory);
+
+		return cachingConnectionFactory;
+	}
+
+	@Bean
+	@ConditionalOnSingleCandidate(ConnectionFactory.class)
+	@ConditionalOnMissingBean(RabbitOperations.class)
+	public RabbitTemplate rabbitTemplate(RabbitProperties rabbitProperties, ConnectionFactory connectionFactory, ObjectProvider<MessageConverter> messageConverter,
```

Comment:
**IGNORE FOR NOW** If the above approach is good for the connection factory, I will do something similar here for the RabbitTemplate.

Reply:
I will get back to this MR in the next 24-48 hrs to follow through w/ my comment above.